drivers/gles3/storage/material_storage.cpp

```diff
@@ -1143,9 +1143,9 @@ MaterialStorage::MaterialStorage() {
 		actions.renames["CANVAS_MATRIX"] = "canvas_transform";
 		actions.renames["SCREEN_MATRIX"] = "screen_transform";
 		actions.renames["TIME"] = "time";
-		actions.renames["PI"] = _MKSTR(Math::PI);
-		actions.renames["TAU"] = _MKSTR(Math::TAU);
-		actions.renames["E"] = _MKSTR(Math::E);
+		actions.renames["PI"] = String::num(Math::PI);
```

Comment:
So previously it assigned `actions.renames["PI"] = "PI"`, now it assigns `actions.renames["PI"] = 3.1415926...`. I think for shaders the former behavior makes more sense, no?

Reply:
Oh wait, it probably did replace via the macro definition.
I guess the new version is somewhat slower still, but maybe that's acceptable?

Reply:
`PI` exists in the Godot shader language but not in GLSL. Some manual definitions of it exist though, such as here:

https://github.com/godotengine/godot/blob/7b9c5122faafb6b4a655e31ae27bcffeb5cf254f/drivers/gles3/shaders/effects/copy.glsl#L162

Reply:
A lot of the operations converting to String could probably get `constexpr` equivalents with `const char *` outputs, or maybe even String *itself* if it had `constexpr` support (probably couldn't happen until c++20), if speed ends up being a concern